### PR TITLE
Fix python feature issue #258, improve tests, fix oryx bug

### PR DESCRIPTION
**Pull request by Chuxel**
As described in #258, currently the "defaultInterpreterPath" is on correct in a number of cases since the code and path in the python feature was modified to support more than one version of python.

I fixed the additional python test that would have returned false positives (the `| grep ...` was for the test command so it always was successful).

Since the feature now manages a set of symlinks, I added test coverage for those since that was also part of the problem with os-provided... no symlink was created.

I then added one for oryx.  That exposed a bug where the target folder may be missing so I fixed that.

### CODE REVIEW

**Comment by Chuxel** [comment]
The Debian 9 test failed since stretch is completely out of support now and Python 3.6 is as well - so none of the tools will install in it.  I'd suggest removing the test and making clear we don't support it in any feature at this point.

Happy to do that in this PR if it makes sense.

**Review by joshspicer — APPROVED** [review]
No comments - this looks great to me, thanks!

**Comment by joshspicer** [comment]
> Happy to do that in this PR if it makes sense.

That would be great, go for it please :)

**Review by samruddhikhandale — APPROVED** [review]
Looks great! Thanks for fixing this and adding test coverage. 

> Happy to do that in this PR if it makes sense.

+1 Makes sense!

File: src/python/install.sh [hunk 1]
```diff
@@ -299,13 +299,16 @@ install_using_oryx() {
...
+    # The python install root path may not exist, so create it
+    mkdir -p "${PYTHON_INSTALL_PATH}"
     oryx_install "python" "${VERSION}" "${INSTALL_PATH}" "lib" || return 1
 
     ln -s "${INSTALL_PATH}/bin/idle3" "${INSTALL_PATH}/bin/idle"
     ln -s "${INSTALL_PATH}/bin/pydoc3" "${INSTALL_PATH}/bin/pydoc"
     ln -s "${INSTALL_PATH}/bin/python3-config" "${INSTALL_PATH}/bin/python-config"
 
-    add_symlink
+   add_symlink
```
[the remark this answers is not in the document]
Comment (Chuxel) [reply src/python/install.sh:311]: Nope!  Will fix.

File: src/python/install.sh [hunk 2]
```diff
@@ -299,6 +299,9 @@ install_using_oryx() {
         echo "(!) Python version ${VERSION} already exists."
         exit 1
     fi
+
+    # The python install root path may not exist, so create it
+    mkdir -p "${PYTHON_INSTALL_PATH}"
```
Comment (samruddhikhandale) [inline src/python/install.sh:304]: Can we bump the patch version in here as we are adding a new `option`?https://github.com/devcontainers/features/blob/cc560e605dbe2f631a66ec9463e19acb728db4e8/src/python/devcontainer-feature.json#L3

The Python Feature will automatically get published once this PR is merged. (Provided the version has changed)

**Pull request merged**